lib/phoenix/router.ex

```diff
@@ -346,32 +346,35 @@ defmodule Phoenix.Router do
 
     case pipeline.(conn) do
       %Plug.Conn{halted: true} = halted_conn ->
+        emit_router_dispatch_end(:stop, conn, start, metadata)
```

Comment:
I would rather avoid the indirection and do the following:

```suggestion
        measurements = %{duration: System.monotonic_time() - start}
        :telemetry.execute([:phoenix, :router_dispatch, :stop], measurements, metadata)
```

Aside from the tests, this should be the only change we need :) Wdyt?

Reply:
Sounds good. My intention wasn't indirection, rather consistency.  :)  there was already a fix for making things consistent in [#4049](https://github.com/phoenixframework/phoenix/pull/4049). There may be a better way to help aid that goal, but since there aren't a lot of things to keep consistent, I agree it's probably better to leave that for a future refactoring if these measurements become more complex.

Updated!

Reply:
Indirection was perhaps the wrong word. What I meant was, in this particular instance, especially for the purposes of review, less is more :) Thank you for the update!